txnkv/txnlock/lock_resolver.go

```diff
@@ -219,9 +219,45 @@ func (lr *LockResolver) getResolved(txnID uint64) (TxnStatus, bool) {
 	return s, ok
 }
 
+// TryBatchResolveLocks tries to resolve locks with batch method.
+func (lr *LockResolver) TryBatchResolveLocks(
+	bo *retry.Backoffer,
+	locks []*Lock,
+	loc locate.RegionVerID,
+	safepoint uint64,
+	targetResolveTS uint64,
+) (bool, error) {
+	if len(locks) == 0 {
+		return true, nil
+	}
+
+	logutil.BgLogger().Info("TryBatchResolveLocks", zap.Uint64("safepoint", safepoint), zap.Uint64("target-resolve-ts", targetResolveTS))
+
+	locksBeforeSafePoint := make([]*Lock, 0, len(locks))
+	locksAfterSafePoint := make([]*Lock, 0, len(locks))
+	for _, l := range locks {
+		if l.TxnID < safepoint {
```

Comment:
```suggestion
		if l.TxnID <= safepoint {
```

To be consistent with the current implementation of scan_lock in tikv, here it's inclusive. It shouldn't matter much though.